site/docs/extensions/index.md

```diff
@@ -0,0 +1,38 @@
+# Extensions
```

Comment:
I'm sorry I've been bad about keeping up with things but do we have concrete examples of extensions that we are trying to support?

I worry that a lot of the structure in the design might be overly broad and without having concrete use-cases might yield sub-optimal approaches?

Reply:
Are you talking about advanced extensions? I was hoping (?) the documentation in this patch better clarified the purposes of simple extensions.

A few advanced extensions are:

- Add some new generic relation optimization information
   - E.g. Add selectivity estimates to improve execution behavior
- Add a new behavior to an existing relation
   - E.g. Create an Aggregate that excludes null keys (something requested in #42)
- Introduce a new relation
   - E.g. A combination of a filter, aggregate and project corresponding to a specific execution engine
- Introduce a new kind of read type
   - E.g. Build a read type for an internal system